specification/maintenance/Maintenance.Management/ConfigurationAssignment.tsp

```diff
@@ -0,0 +1,301 @@
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+import "@typespec/openapi";
+import "@typespec/rest";
+import "./models.tsp";
+
+using TypeSpec.Rest;
+using Azure.ResourceManager;
+using TypeSpec.Http;
+using TypeSpec.OpenAPI;
+
+namespace Microsoft.Maintenance;
+/**
+ * Configuration Assignment
+ */
+#suppress "@azure-tools/typespec-azure-core/no-private-usage"
+#suppress "@azure-tools/typespec-azure-core/composition-over-inheritance"
+@Azure.ResourceManager.Private.armResourceInternal(
+  ConfigurationAssignmentProperties
+)
+@TypeSpec.Http.Private.includeInapplicableMetadataInPayload(false)
+model ConfigurationAssignment extends Foundations.ProxyResource {
```

Comment:
Why can't we just use `ProxyResource<T>` for this, or ExtensionResource<T>?

This is an extension resource, we should not use private decorators unless absolutely necessary.